src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySpark.java

```diff
@@ -0,0 +1,485 @@
+        final String baseQualities = ReadUtils.getBaseQualityString(read);
+        writer.write(new FastqRecord(readName, readBases, "", baseQualities));
+    }
+
+    /**
+     * Final return type of the whole process of SGA local assembly.
+     * assembledContigFile is the file containing the assembled contigs, if the process executed successfully, or null if not.
+     * runtimeInformation contains the runtime information logged along the process up until the process erred, if errors happen,
+     *  or until the last step if no errors occur along the line.
+     *  The list is organized along the process of executing the assembly pipeline, inside each entry of the list is
+     *      the name of the module,
+     *      return status of the module,
+     *      stdout message of the module, if any,
+     *      stderr message of the module, if any
+     */
+    @VisibleForTesting
+    static final class SGAAssemblyResultType implements Serializable{
+        private static final long serialVersionUID = 1L;
+
+        public final File assembledContigFile;
+        public final List<Tuple2<String, Tuple3<Integer, String, String>>> runtimeInformation;
+
+        public SGAAssemblyResultType(final File contigFile, final List<Tuple2<String, Tuple3<Integer, String, String>>> collectedRuntimeInfo){
+            assembledContigFile = contigFile;
+            runtimeInformation = collectedRuntimeInfo;
+        }
+    }
+
+    /**
+     * Performs assembly on the FASTA files. Essentially a wrapper around a limited number of modules in SGA.
+     * @param fastqFilesForEachBreakPoint temporary FASTA file of sequences around the putative breakpoint
+     * @param sgaPath                     full path to SGA
+     * @param threads                     number of threads to use in various modules, if the module supports parallelism
+     * @param runCorrections              user's decision to run sga's corrections (with default parameter values) or not
+     * @return                            contig file and runtime information, associated with the breakpoint id
+     */
+    private static Tuple2<Long, SGAAssemblyResultType> performAssembly(final Tuple2<Long, File> fastqFilesForEachBreakPoint,
+                                                                       final Path sgaPath,
+                                                                       final int threads,
+                                                                       final boolean runCorrections){
+
+        final File rawFASTQ = fastqFilesForEachBreakPoint._2();
+
+        final SGAAssemblyResultType assembledContigsFileAndRuntimeInfo = SGASerialRunner(sgaPath, rawFASTQ, threads, runCorrections);
+
+        return new Tuple2<>(fastqFilesForEachBreakPoint._1(), assembledContigsFileAndRuntimeInfo);
+    }
+
+    @VisibleForTesting
+    static SGAAssemblyResultType SGASerialRunner(final Path sgaPath, final File rawFASTQ, final int threads, final boolean runCorrections){
```

Comment:
If possible, try to make method names start with a lower case letter and be a verb. I thought this was a class for a minute. How about runSGAInSerial()?